api/envoy/config/trace/v3/zipkin.proto

```diff
@@ -68,4 +68,11 @@ message ZipkinConfig {
   // Optional hostname to use when sending spans to the collector_cluster. Useful for collectors
   // that require a specific hostname. Defaults to :ref:`collector_cluster <envoy_v3_api_field_config.trace.v3.ZipkinConfig.collector_cluster>` above.
   string collector_hostname = 6;
+
+  // If this is set to true, then Envoy will be treated as independent hop in trace chain. Server span will be created
+  // for downstream request and client span will be created for upstream request.
+  // If this is set to true, then the
+  // :ref:`start_child_span of router <envoy_v3_api_field_extensions.filters.http.router.v3.Router.start_child_span>`
+  // SHOULD be set to true also to ensure the correctness of trace chain.
+  bool independent_proxy = 7;
```

Comment:
This flag is pretty confusing when used with operation_name and shared_span_context. Would it be better to add an e-num option to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-enum-config-core-v3-trafficdirection instead, which I think is the source of truth for the tracer behavior?

Also, regarding naming: I think gateway or a middle proxy would be a better term here.

Reply:
> Would it be better to add an e-num option to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-enum-config-core-v3-trafficdirection instead, which I think is the source of truth for the tracer behavior?

If we add third value to the enum and set it, we will lose the original direction info of inbound or outbound. It may breaks some other extensions that depend on the traffic direction.

Reply:
> Also, regarding naming: I think gateway or a middle proxy would be a better term here.

`independent_gateway` or `independent_middle_proxy`?

Reply:
We'll probably need to do the same thing for other tracers, no? Other tracers can treat a new enum option as `UNSPECIFIED` which shouldn't break code because traffic direction was never required.

As for the naming, I'll leave it to the API reviewer. The end result is that the proxy becomes visible in the trace graph, so maybe making it imply that would be easier to understand.

Reply:
```
  // Specifies the intended direction of the traffic relative to the local Envoy.
  // This property is required on Windows for listeners using the original destination filter,
  // see :ref:`Original Destination <config_listener_filters_original_dst>`.
  core.TrafficDirection traffic_direction = 16;
```

> Other tracers can treat a new enum option as UNSPECIFIED which shouldn't break code because traffic direction was never required.

I am not mean other tracers. The traffic direction is common attributes of listener, so may affects may other extensions. For example, original dst filter on windows.


> We'll probably need to do the same thing for other tracers, no?

Only tracers that have span type of client/server need this. For example, the skywalking tracer. But different from zipkin, skywalking always treats envoy as independent hop in the trace chain.

Reply:
If it's necessary, it's ok for me to create a common proto API that could be used by all tracers for this situation. But IMO, I think we shouldn't mixed it with `traffic_direction`. 🤔

Reply:
Ok, the original motivation for `traffic_direction` was because every telemetry extension needed this flag, and it caused a lot of redundancy in xDS. We can always revisit it later if it becomes a problem again, so it's fine to add a custom flag just for zipkin.